src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -101,8 +105,14 @@ protected async Task<string> GetObserverResource(string url, string resourceId =
                 request.Headers.TryAddWithoutValidation("Authorization", await GetToken(resourceId));
             }
 
-            var cancelToken = new CancellationToken();
+            var cancelTokenSource = new CancellationTokenSource();
+            var cancelToken = cancelTokenSource.Token;
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
             var response = await _httpClient.SendAsync(request, cancelToken);
```

Comment:
I think I forgot to remove this but can you remove the `cancelToken` as the `DataProviderLogDecorator` class already handles cancellationToken. You can think of the `DataProviderLogDecorator` as the frontend to all data provider calls. It executes the data provider tasks and logs exceptions and helpful summary information such as latency